use ember-auto-import instead of ember-browserify #2

sfbwalder opened the pull request:
This is a rendition of #77.

All tests and step libraries in the parent application/addon import yadda using:
```import { yadda } from 'ember-cli-yadda/test-support/yadda'```

As you can see above, the yadda library is imported into ember-cli-yadda and exported for the parent's use in [ember-cli-yadda/test-support/yadda.js](https://github.com/albertjan/ember-cli-yadda/compare/master...StateFarmIns:auto-import?expand=1#diff-37789e81451a23232ab0cda32821175a).  The import has to be in a module that is in ember-cli-yadda's name space because of how ember-auto-import works (see https://github.com/ef4/ember-auto-import/issues/84).

Importing yadda in the helper module installed in the parent through the ember-cli-yadda blueprint does not work.  The helper ends up in the parent's name space rather than the addon's, so it does not trigger yadda to be auto-imported.  It could be done this way only if the parent app adds yadda as a dependency.

To avoid that and provide backwards compatibility the [yadda helper module in the blueprint](https://github.com/albertjan/ember-cli-yadda/compare/master...StateFarmIns:auto-import?expand=1#diff-20617953073fc32d4f6856f2e63b8f3b) is still there, but it is importing yadda from test-support/yadda instead of using npm:yadda.  `ember g ember-cli-yadda` will need to be run to update the parent's `tests/helpers/yadda.js`.

If the desire is to clean up a bit, then the helper could be deleted from the parent and imports of yadda from 'tests/helpers/yadda' updated to pull from test-support yadda module.

Also, take a look at the [autoImport config in index.js](https://github.com/albertjan/ember-cli-yadda/pull/79/files#diff-168726dbe96b3ce427e7fedce31bb0bc).  I am not familiar with webpack, but that seemed to work.

Any one have ideas for other approaches to tighten this up further?

sfbwalder commented on index.js:
```diff
@@ -5,6 +5,16 @@ const FeatureParser = require('./lib/feature-parser');
   name: 'ember-cli-yadda',
+  options: {
+    autoImport:{
+      exclude: ['mocha'],
```
This avoids the build error:
```
ember-cli-yadda tried to import "mocha" from addon code, but "mocha" is a devDependency. You may need to move it into dependencies.
```

sfbwalder commented on index.js:
```diff
@@ -5,6 +5,16 @@ const FeatureParser = require('./lib/feature-parser');
+      exclude: ['mocha'],
+      webpack: {
+        node: {
+          fs: "empty"
```
This avoids a set of errors similar to:
```
ERROR in ./node_modules/yadda/lib/shims/index.js
Module not found: Error: Can't resolve 'fs' in 'C:\dev\ember\ember-cli-yadda\node_modules\yadda\lib\shims'
 @ ./node_modules/yadda/lib/shims/index.js 12:20-33
 @ ./node_modules/yadda/lib/index.js
 @ ./tmp/ember_auto_import_webpack-staging_dir-gOibSogV.tmp/tests.js
```

simonihmig commented:
For the record: this branch was working fine with `ember-auto-import` 1.12.10, but is failing for 1.12.11 with `Could not find module `yadda` imported from `ember-cli-yadda/test-support/yadda`. Likely a bug in that version of ember-auto-import though...

simonihmig commented:
The `ember-auto-import` issue has been resolved with 1.12.13, which we should update to. There is also another one we seem to have to fix here: https://github.com/ef4/ember-auto-import/issues/111#issuecomment-415660116.

Besides that, any thoughts on my remarks above, @sfbwalder?  I would love to land this sooner that later... 😀

sfbwalder commented:
**Updates**

> Maybe it's better to just remove the file, and tell users to update their imports? 

Yes.  I removed test/helpers/yadda.js

> Another somewhat "better looking" possibility would be to remove the test-support namespace...

Yes.  I followed the pattern you linked to, and now it is:
```
import { yadda } from 'ember-cli-yadda';
```

> The `ember-auto-import` issue has been resolved with 1.12.13, which we should update to.

Done.  Changed the [included hook call to super](https://github.com/albertjan/ember-cli-yadda/pull/79/files#diff-168726dbe96b3ce427e7fedce31bb0bcL31) as suggested [by @ef4](https://github.com/ef4/ember-auto-import/issues/111#issuecomment-415660116)

**This will be a breaking change.**  I bumped the version to 0.5.0.

**To upgrade** one of our apps to use this version I did the following:
1.  npm install ember-cli-yadda@0.5.0 --save-dev
1. npm uninstall ember-browserify
1. Global find/replace of `import yadda from '../../helpers/yadda';` with `import { yadda } from 'ember-cli-yadda';`
1. Delete `tests/helpers/yadda.js`
1. Ran the tests - all successful

sfbwalder commented:
How do you want to handle upgrade instructions for moving to 0.5.0?  

Initially I thought release notes would be sufficient, but readme.md has a section on upgrading to 0.4.0 that needs something done with it.  

Should we split it out, perhaps like this:
* Upgrading from 0.4.0 to 0.5.0
* Upgrading from <0.4.0 to 0.5.0

Or move the current readme.md to /docs, and only have 0.4.0 to 0.5.0?  
Or remove the upgrade section and point to release notes?

simonihmig commented:
> To upgrade one of our apps to use this version I did the following:

Cool, I'll try the same with one of our bigger apps on Monday! 

> Or remove the upgrade section and point to release notes?

Sounds good to me! The current instructions can be added to the 0.4.0 release notes, and the new ones to the upcoming 0.5.0. With links from Readme. I think that would be fine!

simonihmig commented:
@sfbwalder I tried your branch on our app, and everything went fine! 🎉

sfbwalder commented:
I adjusted the readme and release notes, published, and tested by installing 0.5.0 from public npm.  Everything worked.  Please adjust the release notes or let me know what you would like changed.

Otherwise, I think this change is complete.

simonihmig commented:
Looks perfect, thanks for working on this! 👍
